starter-code/src/viking.js

```diff
@@ -1,11 +1,98 @@
 // Soldier
-function Soldier () {}
+function Soldier (health, strength) {
+    this.health = health;
+    this.strength = strength;
+
+    this.attack = function(){
```

Comment:
Don't define the object's methods inside the constructor. Instead, outside it, do:
```
Soldier.prototype.attack = function () {
  return this.strength;
};
```

Reply:
Why is that? 
Wouldn´t this make more susceptible to error-prone?

Reply:
Mainly for performance and economy. We talked about this when you were doing the exercise. Of course, we can discuss it again. Anyways, here you have more info: https://stackoverflow.com/questions/4508313/advantages-of-using-prototype-vs-defining-methods-straight-in-the-constructor